frontend/src/pages/articles/libs/helpers/get-article-tags.ts

```diff
@@ -0,0 +1,27 @@
+import { type ArticleWithAuthorType } from '~/packages/articles/articles.js';
+
+import { type TagType } from '../types/types.js';
+
+const getArticleTags = (article: ArticleWithAuthorType): TagType[] => {
+  const result: TagType[] = [];
+  const { prompt, genre } = article;
+
+  if (genre) {
+    result.push({
+      category: 'genre',
+      text: genre,
+    });
+  }
+  if (prompt) {
+    let keys: keyof typeof prompt;
+    for (keys in prompt) {
+      const value = prompt[keys];
+      if (value) {
+        result.push({ category: keys, text: value });
+      }
+    }
+  }
+  return result;
+};
```

Comment:
Implemetation changed because Object.entries no longer possible to use because of type issues.  That's why I submitted new variant and awaiting for review.

Reply:
Fact that `key is string` on Object.entries does not mean that you have no ability to cast it to needed type - if you know that key actually would be of that type

Reply:
I thought you are not supposed to do it if possible because it defeats the purpose of typescript.

Reply:
also we have some linter problems with this variant 
![image](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/44391531/b4e2b56e-a4c8-4a06-bbc9-da01d225b5a4)

Reply:
Yep, about no-array-reduce -- I've mentioned in previous comments that it may be disabled

Reply:
About TS - you are using technology, this thesis should not work vice versa 🥲

Reply:
Well , it's a good point. I am just trying to understand why reduce is better here.
Should I disable prefer-spread rule aswell ?

Reply:
Spread can also be disabled, yes
Because with reduce you're taking new instance of entity without mutating it on any steps + reduce works with arrays that means we won't provide confusion with iterating through object (like it could be with for...in) or anything else

Reply:
Interesting , thank you.